src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -1602,4 +1602,46 @@ HRESULT FunctionMethodSignature::TryParse() {
   return S_OK;
 }
 
+bool FindTypeDefByName(
+    const trace::WSTRING instrumentationTargetMethodTypeName,
+    const trace::WSTRING assemblyName,
+    const ComPtr<IMetaDataImport2>& metadata_import, mdTypeDef& typeDef) {
+  mdTypeDef parentTypeDef = mdTypeDefNil;
+  auto nameParts = Split(instrumentationTargetMethodTypeName, '+');
+  auto instrumentedMethodTypeName = instrumentationTargetMethodTypeName;
+
+  if (nameParts.size() == 2) {
+    // We're instrumenting a nested class, find the parent first
+    auto hr = metadata_import->FindTypeDefByName(nameParts[0].c_str(),
+                                                 mdTokenNil, &parentTypeDef);
+
+    if (FAILED(hr)) {
+      // This can happen between .NET framework and .NET core, not all apis are
+      // available in both. Eg: WinHttpHandler, CurlHandler, and some methods in
+      // System.Data
+      Debug("Can't load the parent TypeDef: ", nameParts[0],
+            " for nested class: ", instrumentationTargetMethodTypeName,
+            ", Module: ", assemblyName);
+      return false;
+    }
+    instrumentedMethodTypeName = nameParts[1];
+
+  } else if (nameParts.size() > 1) {
```

Comment:
I know this works, but I was why >1? is >2.

Reply:
I agree. The case where `== 2` is already handled above so this is confusing.
```suggestion
  } else if (nameParts.size() > 2) {
```

Reply:
For the back story, it's because I incorrectly had `==1` as the first case initially and forgot to change this one when I fixed it 😬 

I'll leave my badge and gun/keyboard at reception.

Reply:
😂